File: problem1/problem1.cpp
#include <iostream>
#include <stack>
using namespace std;


stack<int>* decToBin(int decimal){
    stack<int>* bin = new stack<int>;
    while (decimal != 0){
        int remain = decimal % 2;
        bin->push(remain);
        decimal /= 2;
    }
    return bin;
}

int main(){
    stack<int>* binary;
    int decimal;

    cout << "Enter your decimal number: ";
    cin >> decimal;
    binary = decToBin(decimal);

    while (!binary->empty()){
        cout << binary->top() << " ";
        binary->pop();
    }
    return 0;
}

File: problem1/problem1.py
def decToBin(decimal:int) -> list[int]:
    binary = []

    while (decimal != 0):
        remain = decimal % 2
        binary.append(remain)
        decimal = decimal // 2
    
    return binary


if __name__ == "__main__":
    decimal:int = int(input("Enter your decimal number: "))
    binary = decToBin(decimal)
    while len(binary) != 0:
        print(binary.pop(), end=" ")



